src/common/target_data.c

```diff
@@ -120,6 +120,89 @@ xdd_init_new_target_data(target_data_t *tdp, int32_t n) {
 
 } /* end of xdd_init_new_target_data() */
 
+/*----------------------------------------------------------------------------*/
+/* xdd_set_bs_reqsize_numreqs_from_loadfile() - This function reads a target's
+ * seek load file and sets the block size, request size (override the values
+ * from command line) and numreqs if td_numreqs and td_bytes are not provided.
+ */
+static int32_t
+xdd_set_bs_reqsize_numreqs_from_loadfile(target_data_t *tdp) {
+	FILE	*loadfp;			/* Load File Point */
+	char	line[LINE_LENGTH];	/* one line of characters */
+	struct seekhdr	*sp;
+
+	sp = &tdp->td_seekhdr;
+	/* Open the load file */
+	loadfp = fopen(sp->seek_loadfile, "r");
+	if (loadfp == NULL) {
+		fprintf(xgp->errout, "%s: ERROR: BLock size not found in the first line of %s\n",
+				xgp->progname, sp->seek_loadfile);
+		fclose(loadfp);
+		return(-1);
+	}
+
+	/* Read the first line */
+	if (fgets(line, sizeof(line), loadfp) == NULL) {
```

Comment:
Why `fgets` and not `getline`?

Reply:
Will `getline` have compatibility issue?